bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/handler/PulseaudioHandler.java

```diff
@@ -218,67 +213,25 @@ public void dispose() {
-            deviceOnlineWatchdog();
-        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE
+        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE
                 || bridgeStatusInfo.getStatus() == ThingStatus.UNKNOWN) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.UNINITIALIZED) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
         }
     }
 
-    private void deviceOnlineWatchdog() {
-        Runnable runnable = () -> {
-            try {
-                PulseaudioBridgeHandler bridgeHandler = getPulseaudioBridgeHandler();
-                if (bridgeHandler != null) {
-                    if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
-                        if (bridgeHandler.getDevice(name) == null) {
-                            updateStatus(ThingStatus.OFFLINE);
-                            this.bridgeHandler = null;
-                        } else {
-                            updateStatus(ThingStatus.ONLINE);
-                        }
-                    } else {
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
-                    }
-                } else {
-                    logger.debug("Bridge for pulseaudio device {} not found.", name);
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
-                }
-            } catch (Exception e) {
-                logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
-                this.bridgeHandler = null;
-            }
-        };
-
-        refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, refresh, TimeUnit.SECONDS);
-    }
-
     private synchronized @Nullable PulseaudioBridgeHandler getPulseaudioBridgeHandler() {
-        if (this.bridgeHandler == null) {
-            Bridge bridge = getBridge();
-            if (bridge == null) {
-                logger.debug("Required bridge not defined for device {}.", name);
-                return null;
-            }
-            ThingHandler handler = bridge.getHandler();
-            if (handler instanceof PulseaudioBridgeHandler) {
-                this.bridgeHandler = (PulseaudioBridgeHandler) handler;
-                this.bridgeHandler.registerDeviceStatusListener(this);
-            } else {
-                logger.debug("No available bridge handler found for device {} bridge {} .", name, bridge.getUID());
-                return null;
-            }
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            return null;
+        }
+        ThingHandler handler = bridge.getHandler();
+        if (handler instanceof PulseaudioBridgeHandler) {
+            return (PulseaudioBridgeHandler) handler;
+        } else {
+            throw new IllegalStateException("bridge Handler is of the wrong type");
```

Comment:
`handler` could be null if the bridge handler is not yet initialized.

Reply:
Fixed by changing the log to "bridge Handler is null or of the wrong type"

Reply:
I am not sure at all that having a null bridge handler is an illegal state. It can happen at startup in case your thing handler is initialized before your bridge handler. So you should IMHO return null, like it was done before.

Reply:
OK. I can't remember why I choosed to change this method. Probably because of my work on the nullable annotation, I suppose I tried to make this method returns NonNull and find it impossible along the way ?
Anyway, it was not necessary and I return like it was done before.